wellcomecollection/catalogue-pipeline
# Relation embedder lambda infra build

## What does this change?

Part of https://github.com/wellcomecollection/catalogue-pipeline/issues/2770
Adds relation_emb lambda to the build script, terraform modules and the shell of a relation_emb lambda handler 

## How to test

Build the image, eg. `docker build --target lambda_rie -t lambda_relation .`
Run `docker run -p 9000:8080 lambda_relation`
Run `cat scripts/batches.txt | python scripts/post_to_rie.py` and see the lambda getting hit

TF plan `Plan: 13 to add, 2 to change, 0 to destroy.` 
13 to add are all relation_embedder_lambda resources
2 to update:
```
Terraform will perform the following actions:

  # module.pipeline.module.inference_capacity_provider.aws_autoscaling_group.asg will be updated in-place
  ~ resource "aws_autoscaling_group" "asg" {
        id                               = "catalogue-2024-11-05_inferrer_asg"
        name                             = "catalogue-2024-11-05_inferrer_asg"
        # (31 unchanged attributes hidden)

      ~ launch_template {
            id      = "lt-0ae23e659ef16ca3a"
            name    = "catalogue-2024-11-05_inferrer_launch_template"
          ~ version = "3" -> (known after apply)
        }

        # (2 unchanged blocks hidden)
    }

  # module.pipeline.module.inference_capacity_provider.aws_launch_template.launch_template will be updated in-place
  ~ resource "aws_launch_template" "launch_template" {
      ~ default_version                      = 3 -> (known after apply)
        id                                   = "lt-0ae23e659ef16ca3a"
      ~ image_id                             = "ami-082bbd3675c6d658c" -> "ami-09e2d602becc5f1fa"
      ~ latest_version                       = 3 -> (known after apply)
        name                                 = "catalogue-2024-11-05_inferrer_launch_template"
        tags                                 = {}
        # (16 unchanged attributes hidden)

        # (3 unchanged blocks hidden)
    }

```

## How can we measure success?

We can deploy a relation_embedder_lambda alongside the service

## Have we considered potential risks?

Very low to inexistent
